app/commands/v2/unpublish.rb

```diff
@@ -0,0 +1,137 @@
+module Commands
+  module V2
+    class Unpublish < BaseCommand
+      def call
+        content_id = payload.fetch(:content_id)
+        content_item = find_unpublishable_content_item(content_id)
+
+        unless content_item.present?
+          message = "Could not find a content item to unpublish"
+          raise_command_error(404, message, fields: {})
+        end
+
+        check_version_and_raise_if_conflicting(content_item, previous_version_number)
+
+        if draft_present?(content_id)
+          if payload[:discard_drafts] == true
+            DiscardDraft.call(
+              {
+                content_id: content_id,
+                locale: locale,
+              },
+              downstream: downstream,
+            )
+          else
+            message = "Cannot unpublish with a draft present"
+            raise_command_error(422, message, fields: {})
+          end
+        end
+
+        case type = payload.fetch(:type)
+        when "withdrawal"
+          withdraw(content_item)
+        when "redirect"
+          redirect(content_item)
+        when "gone"
+          gone(content_item)
+        else
+          message = "#{type} is not a valid unpublishing type"
+          raise_command_error(422, message, fields: {})
+        end
+
+        Success.new(content_id: content_id)
+      end
+
+    private
+
+      def withdraw(content_item)
+        unpublishing = State.unpublish(content_item,
+          type: "withdrawal",
+          explanation: payload.fetch(:explanation),
+        )
+
+        send_content_item_downstream(content_item, unpublishing) if downstream
+      end
+
+      def redirect(content_item)
+        unpublishing = State.unpublish(content_item,
+          type: "redirect",
+          alternative_path: payload.fetch(:alternative_path),
+        )
+
+        redirect = RedirectPresenter.present(
+          base_path: Location.find_by(content_item: content_item).base_path,
```

Comment:
It doesn't make sense to unpublish *any* content without a location, so I've added checks for that.